plugin/trino-mongodb/src/test/java/io/trino/plugin/mongodb/BaseMongoConnectorTest.java

```diff
@@ -104,6 +104,39 @@ public void testSortItemsReflectedInExplain()
                 "TopNPartial\\[5 by \\(nationkey DESC");
     }
 
+    @Test(dataProvider = "guessFieldTypesProvider")
+    public void testGuessFieldTypes(String mongoValue, String trinoValue)
+    {
+        Document document = Document.parse(format("{\"test\":%s}", mongoValue));
+
+        assertUpdate("DROP TABLE IF EXISTS test.test_guess_field_type");
+        client.getDatabase("test").getCollection("test_guess_field_type").insertOne(document);
+
+        assertThat(query("SELECT test FROM test.test_guess_field_type"))
+                .matches("SELECT " + trinoValue);
+
+        assertUpdate("DROP TABLE test.test_guess_field_type");
+    }
+
+    @DataProvider
+    public Object[][] guessFieldTypesProvider()
+    {
+        return new Object[][] {
```

Comment:
Would a `null` value cause issues with guessing?

Maybe we should add a test to document the existing behaviour?

Reply:
`null` would be simply handled as unknown types.